Mage/src/main/java/mage/game/command/emblems/EmblemOfCard.java

```diff
@@ -0,0 +1,65 @@
+package mage.game.command.emblems;
+
+import mage.MageObject;
+import mage.abilities.AbilityImpl;
+import mage.cards.Card;
+import mage.cards.decks.DeckCardInfo;
+import mage.cards.decks.importer.CardLookup;
+import mage.cards.mock.MockCard;
+import mage.constants.Zone;
+import mage.game.command.Emblem;
+import org.apache.log4j.Logger;
+
+import java.util.stream.Collectors;
+
+public final class EmblemOfCard extends Emblem {
+    private static final Logger logger = Logger.getLogger(EmblemOfCard.class);
+    public static Card cardFromDeckInfo(DeckCardInfo info) {
+        return CardLookup.instance.lookupCardInfo(info.getCardName(), info.getSetCode(), info.getCardNum())
+                .orElseThrow(() -> new IllegalArgumentException("No real card for DeckCardInfo" + info.getCardName()))
+                .getCard();
+    }
+    public EmblemOfCard(Card card, Zone zone) {
+        super(card.getName());
+        if (card instanceof MockCard) {
+            MockCard mock = (MockCard)card;
+            card = CardLookup.instance.lookupCardInfo(card.getName(), card.getExpansionSetCode(), card.getCardNumber())
+                    .orElseThrow(() -> new IllegalArgumentException("No real card for mock card " + mock.getFullName(true)))
+                    .getCard();
+        }
+        this.getAbilities().addAll(card.getAbilities().stream().filter(
+                ability -> zone.match(ability.getZone())
+        ).map(ability -> {
+            if (ability instanceof AbilityImpl && ability.getZone() == zone) {
+                return ((AbilityImpl)ability).copyWithZone(Zone.COMMAND);
+            }
+            return ability;
+        }).collect(Collectors.toList()));
+        this.getAbilities().setSourceId(this.getId());
+        this.setExpansionSetCode(card.getExpansionSetCode());
+        this.setCardNumber(card.getCardNumber());
+        this.setImageNumber(card.getImageNumber());
+    }
+    public EmblemOfCard(Card card) {
+        this(card, Zone.BATTLEFIELD);
+    }
+
+    private EmblemOfCard(EmblemOfCard eoc) {
+        super(eoc);
+    }
+    @Override
+    public Emblem copy() {
+        return new EmblemOfCard(this);
+    }
+
+    @Override
+    public void setSourceObject(MageObject sourceObject) {
+        try {
+            super.setSourceObject(sourceObject);
+        }
+        catch (IllegalArgumentException e) {
+            // happens because this isn't a real emblem, but the source object gets set before throwing so it's fine
```

Comment:
If you look at the code for `Emblem.setSourceObject`, the first step is "set the source object", and that happens just fine; the step that fails is "look up the printed token this emblem corresponds to for art purposes". I'd do the first step directly, but `sourceObject` in `Emblem` is `private`. (If you're fine with it changing to `protected`, I could just have this override set the field directly rather than calling the `super` method.)

Reply:
BTW images must be workable too for card emblems. If you need override `setSourceObject` then fill free to improve it.

Reply:
I'm confused as to why images work on normal emblems but not these. I call all the same setters (`setExpansionSetCode`, `setCardNumber`, `setImageNumber`) in the constructor that `Emblem.setSourceObject()` calls.

Reply:
Looking into it further. It seems that the `ImageCache` lookup key for normal emblem images looks something like `"Emblem Ajani#M19#0##<imageSize>` (presumably they're stored in a way that corresponds to this). `EmblemView` completely discards image number and card number, and `CardView` fills them back in with `0`/`""` for emblems; this works fine for the emblem images, but means the actual card images don't get found.

The thing confusing me here is, why does `Emblem.setSourceObject()` bother setting the image number if it's discarded?

Reply:
yup, passing those along for `EmblemOfCard`s specifically fixes images
![image](https://github.com/magefree/mage/assets/19716820/41c7106a-75e0-4d28-a88a-1ff08dfdd5e7)

Reply:
...hmm, still not working for cards with multiple arts, even after passing along `usesVariousArt`

Reply:
ah, missed the copy constructor

Reply:
okay, now that art's done, on the subject of the source object itself
I see two options besides what's done here:
- have `EmblemOfCard` maintain its own `sourceObject` field, *leaving its `Emblem.sourceObject` null*, and override `getSourceObject` and `getSourceId` to use it
  - This works for now, but if anyone ever makes a new `Emblem` method that touches `sourceObject` it'll use the null one and cause all sorts of hard-to-debug problems
- change `Emblem.sourceObject` to `protected` and set it without the `super` call
  - This is more robust, but the exception in `Emblem.setSourceObject` exists for a reason: it tells people they need to add the emblem to the tokens database. If any emblem implementer can just set `sourceObject` directly, there's a risk of them dealing with the error that way rather than the proper way. 

I think the second option might be correct, though; on further study, `setSourceObject` is only called inside `GameImpl.addEmblem`, so card implementers are unlikely to remove the call; and they're unlikely to solve the exception by overriding the method when there's a message saying "how to fix".